xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentsinline.vm

```diff
@@ -153,12 +154,12 @@ $xwiki.ssfx.use('uicomponents/viewers/comments.css', true)
       <div class="btn-group commenttools">
       #if($xwiki.hasAccessLevel('comment'))
         <a class="commentreply btn btn-default btn-xs" rel="nofollow" href="$xredirect.replaceAll('&?replyto=\d++', '')&amp;replyto=${comment.number}#xwikicomment_${comment.number}" title="$services.localization.render('core.viewers.comments.reply')"#if("$!replyTo" == "${comment.number}") style="display: none;"#end>$services.icon.renderHTML('comment')</a></span>
-        #if($services.model.resolveDocument($comment.author, 'current') == $xcontext.userReference || $hasAdmin)
+        #if($hasAdmin || $isUserComment)
           <a class="edit btn btn-default btn-xs" rel="nofollow" href="$doc.getURL('view', "viewer=comments&amp;number=${comment.number}&amp;xredirect=$doc.getURL('view')")" title="$services.localization.render('core.viewers.comments.edit')">$services.icon.renderHTML('pencil')</a>
         #end
       #end
       <a class="permalink btn btn-default btn-xs" rel="nofollow" href="$doc.getURL('view', 'viewer=comments')#xwikicomment_${comment.number}" title="$services.localization.render('core.viewers.comments.permalink')">$services.icon.renderHTML('link')</a>
-      #if ($hasAdmin || (!$isGuest && $hasEdit && $comment.author == $xcontext.user))
+      #if ($hasAdmin || ($hasEdit && $isUserComment))
```

Comment:
You dropped the ``!$isGuest`` condition, probably thinking that if a guest user can add a comment/annotation then he should be able to remove it, but then a guest user will be able to delete the comments/annotations of other guests.

Reply:
I though a guest was a non-logged user. In this case he wouldn't have rights to make annotations and the `$isUserComment` would always be `false`.

Reply:
Well I guess it also depends on the rights that have been setup.

Reply:
AFAIK you can give edit right to guest users. For instance @vmassol allows guest comments on [his blog](http://massol.myxwiki.org/xwiki/bin/view/Blog/).

Reply:
Tried it and the `$isUserComment` is false because the `$comment.author` is empty and $xcontext.userReference`  is null.

Reply:
Which means you're not resolving the comment author correctly. If the comment was added by a guest then the resolved user reference should be null (same as ``$xcontext.userReference``).

Reply:
Right, this behavior has been here since a while (for the remove button on the annotation) but I guess we can fix it now.